mypy/messages.py

```diff
@@ -992,6 +998,34 @@ def pretty_or(args: List[str]) -> str:
     return ", ".join(quoted[:-1]) + ", or " + quoted[-1]
 
 
+def append_invariance_notes(notes: List[str], arg_type: Instance,
+                            expected_type: Instance) -> List[str]:
+    """Explain that the type is invariant and give notes for how to solve the issue."""
+    from mypy.subtypes import is_subtype
+    invariant_type = ''
+    covariant_suggestion = ''
+    if (arg_type.type.fullname() == 'builtins.list' and
+            expected_type.type.fullname() == 'builtins.list' and
+            is_subtype(arg_type.args[0], expected_type.args[0])):
+        invariant_type = 'List'
+        covariant_suggestion = 'Consider using "Sequence" instead, which is covariant'
+    elif (arg_type.type.fullname() == 'builtins.dict' and
+          expected_type.type.fullname() == 'builtins.dict' and
+          isinstance(arg_type.args[0], Instance) and
+          isinstance(expected_type.args[0], Instance) and
+          arg_type.args[0].type == expected_type.args[0].type and
```

Comment:
Instead of this construction of three lines you can just use ``sametypes.is_same_types`` (if I remember the name correctly). This will allow other types, (like ``Union`` etc.) as well, and it is simpler.